db/migrate/20240729103525_update_proxy_byos_column_type.rb

```diff
@@ -1,11 +1,12 @@
 class UpdateProxyByosColumnType < ActiveRecord::Migration[6.1]
   def up
     add_column :systems, :proxy_byos_mode, :integer, default: 0
-    System.where(proxy_byos: false).in_batches.update_all proxy_byos_mode: 0
-    System.where(proxy_byos: true).in_batches.update_all proxy_byos_mode: 1
+    System.where('proxy_byos = 0 and instance_data IS NOT NULL').in_batches.update_all proxy_byos_mode: :payg
```

Comment:
Consider `System.where(proxy_byos: false).where.not(instance_data: nil)` for consistency.

Reply:
good point